source/val/validation_state.h

```diff
@@ -255,6 +260,9 @@ class ValidationState_t {
   /// IDs that are entry points, ie, arguments to OpEntryPoint.
   std::vector<uint32_t> entry_points_;
 
+  /// Functions IDs that are target of OpFunctionCall.
+  std::vector<uint32_t> function_call_targets_;
```

Comment:
It would be better to make this an unordered_map.  We don't want repeated entries if there are several calls to the same function.  We want a fast lookup instead of a linear lookup.   We want O(n+m) time when there are n calls and m function definitions but instead using a std::vector gives us O(n**2 + m).